fortran/nf_nc4.F90

```diff
@@ -971,38 +1023,67 @@ End Function nf_inq_compound_fieldndims
  Function nf_inq_compound_fielddim_sizes( ncid, xtype, fieldid, dim_sizes) &
                                           RESULT (status)
 
-! inq compound field dimension sizes 
+! inq compound field dimension sizes
 
  USE netcdf4_nc_interfaces
 
+ USE ISO_C_BINDING, ONLY: C_INT
+
  Implicit NONE
 
  Integer, Intent(IN)    :: ncid, xtype, fieldid
  Integer, Intent(INOUT) :: dim_sizes(*)
 
  Integer                :: status
 
- Integer(C_INT) :: cncid, cxtype, cfieldid, cstatus 
+ Integer(C_INT) :: cncid, cxtype, cfieldid, cndims, cstatus
+ Integer(C_INT), ALLOCATABLE :: cdim_sizes(:)
+ Integer :: ndims
 
  cncid    = ncid
  cxtype   = xtype
- cfieldid = fieldid - 1 
+ cfieldid = fieldid - 1
+ dim_sizes(1) = 0
```

Comment:
Other than the function name changes, what is happening here? What are the tests you are adding?

I always love to see more test code. What's this all about?

Reply:
We always exchange data with a C function using ISO_C_BINDING variables.  In this case, we allocate a C_INT variable to retrieve cdim_sizes.